pkg/reconcilermanager/controllers/rootsync_controller_test.go

```diff
@@ -634,6 +634,85 @@ func TestRootSyncCreateWithNoSSLVerify(t *testing.T) {
 	t.Log("Deployment successfully created")
 }
 
+func TestUpdateRootReconcilerLogLevelWithOverride(t *testing.T) {
+	// Mock out parseDeployment for testing.
+	parseDeployment = parsedDeployment
+
+	rs := rootSync(rootsyncName, rootsyncRef(gitRevision), rootsyncBranch(branch), rootsyncSecretType(GitSecretConfigKeySSH), rootsyncSecretRef(rootsyncSSHKey))
+	reqNamespacedName := namespacedName(rs.Name, rs.Namespace)
+	fakeClient, fakeDynamicClient, testReconciler := setupRootReconciler(t, rs, secretObj(t, rootsyncSSHKey, configsync.AuthSSH, v1beta1.GitSource, core.Namespace(rs.Namespace)))
+
+	// Test creating Deployment resources.
+	ctx := context.Background()
+	if _, err := testReconciler.Reconcile(ctx, reqNamespacedName); err != nil {
+		t.Fatalf("unexpected reconciliation error, got error: %q, want error: nil", err)
+	}
+
+	rootContainerEnvs := testReconciler.populateContainerEnvs(ctx, rs, rootReconcilerName)
+
+	rootDeployment := rootSyncDeployment(rootReconcilerName,
+		setServiceAccountName(rootReconcilerName),
+		secretMutator(rootsyncSSHKey),
+		containerEnvMutator(rootContainerEnvs),
+		setUID("1"), setResourceVersion("1"), setGeneration(1),
+	)
+	wantDeployments := map[core.ID]*appsv1.Deployment{core.IDOf(rootDeployment): rootDeployment}
+
+	if err := validateDeployments(wantDeployments, fakeDynamicClient); err != nil {
+		t.Errorf("Deployment validation failed. err: %v", err)
+	}
+	if t.Failed() {
+		t.FailNow()
+	}
+	t.Log("Deployment successfully created")
+
+	// Test overriding the log verbosity level of the reconciler, hydration-controller, and git-sync container
+	overrideAllContainerResources := []v1beta1.ContainerSpec{
+		{
+			ContainerName: reconcilermanager.Reconciler,
+			LogLevel:      5,
+		},
+		{
+			ContainerName: reconcilermanager.HydrationController,
+			LogLevel:      7,
+		},
+		{
+			ContainerName: reconcilermanager.GitSync,
+			LogLevel:      9,
+		},
+	}
+
+	if err := fakeClient.Get(ctx, client.ObjectKeyFromObject(rs), rs); err != nil {
+		t.Fatalf("failed to get the root sync: %v", err)
+	}
+	rs.Spec.Override = &v1beta1.OverrideSpec{
+		Specs: overrideAllContainerResources,
+	}
+	if err := fakeClient.Update(ctx, rs); err != nil {
+		t.Fatalf("failed to update the root sync request, got error: %v, want error: nil", err)
+	}
+
+	if _, err := testReconciler.Reconcile(ctx, reqNamespacedName); err != nil {
+		t.Fatalf("unexpected reconciliation error upon request update, got error: %q, want error: nil", err)
+	}
+
+	rootDeployment = rootSyncDeployment(rootReconcilerName,
+		setServiceAccountName(rootReconcilerName),
+		secretMutator(rootsyncSSHKey),
+		containerResourcesMutator(overrideAllContainerResources),
+		containerEnvMutator(rootContainerEnvs),
+		setUID("1"), setResourceVersion("2"), setGeneration(2),
+	)
+	wantDeployments[core.IDOf(rootDeployment)] = rootDeployment
+
+	if err := validateDeployments(wantDeployments, fakeDynamicClient); err != nil {
```

Comment:
Is this actually validating that the log level was set? I don't see anything updating the expected deployment or the validator to assert the log level.

Reply:
the function `validateDeployments` in the unit test have been updated to validate for args. All other unit test container template are also updated to include default log verbosity